app/frontend/src/features/communities/CommunityBase.tsx

```diff
@@ -0,0 +1,131 @@
+import { Breadcrumbs, Link as MuiLink } from "@material-ui/core";
+import { TabContext } from "@material-ui/lab";
+import Alert from "components/Alert";
+import CircularProgress from "components/CircularProgress";
+import TabBar from "components/TabBar";
+import {
+  COMMUNITY_TABS_A11Y_LABEL,
+  communityTabBarLabels,
+  ERROR_LOADING_COMMUNITY,
+  INVALID_COMMUNITY_ID,
+} from "features/communities/constants";
+import { useCommunity } from "features/communities/hooks";
+import { Community } from "pb/communities_pb";
+import { CommunityParent } from "pb/groups_pb";
+import React from "react";
+import { Link, useHistory, useParams } from "react-router-dom";
+import { CommunityTab, routeToCommunity } from "routes";
+import makeStyles from "utils/makeStyles";
+
+import HeaderImage from "./CommunityPage/HeaderImage";
+
+export const useCommunityBaseStyles = makeStyles((theme) => ({
+  root: {
+    marginBottom: theme.spacing(2),
+  },
+  center: {
+    display: "block",
+    marginLeft: "auto",
+    marginRight: "auto",
+  },
+  header: {
+    marginBottom: theme.spacing(1),
+  },
+  breadcrumbs: {
+    "& ol": {
+      justifyContent: "flex-start",
+    },
+  },
+}));
+
+interface CommunityBaseProps {
+  children(communityParams: {
+    community: Community.AsObject;
+    communitySlug?: string;
+  }): React.ReactElement;
+  communityId?: number;
+  defaultTab?: CommunityTab;
+}
+
+export default function CommunityBase({
+  children,
+  communityId,
+  defaultTab,
+}: CommunityBaseProps) {
+  const classes = useCommunityBaseStyles();
+
+  const { communityId: communityIdFromUrl, communitySlug } = useParams<{
+    communityId: string;
+    communitySlug?: string;
+  }>();
+
+  const {
+    isLoading: isCommunityLoading,
+    error: communityError,
+    data: community,
+  } = useCommunity(communityId ?? +communityIdFromUrl);
+
+  const history = useHistory();
+
+  const { page = "overview" } = useParams<{ page: string }>();
+
+  // Use default tab from prop, otherwise derived state from community URL
+  const tab =
+    defaultTab ??
```

Comment:
Material UI was throwing errors when a tab is unselected by passing it `undefined` (type error on `TabContext`) or an unknown value like `unselected` (works, but throws console error on `unselected` being not in the list of tabs), which kinda makes sense as I think tabs aren't designed to be unselected. Since the discussion page won't have a slug, I decided to allow this to be configurable by props too to select a "default tab" to be chosen